### PR TITLE
Add com.authormore.penpotdesktop

### PR DESCRIPTION
**Pull request by Beryesa**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.  
A desktop client for the [Penpot](https://penpot.app/) application [from the community](https://community.penpot.app/t/project-devojo/1468/113?u=beryesa)
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/author-more/penpot-desktop/pull/16 & [thread](https://community.penpot.app/t/project-devojo/1468/112?u=beryesa)

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: com.authormore.Penpot-Desktop.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      # Build the app; in this example the `build` script
+      # in package.json runs electron-builder
+      - |
+        . ../flatpak-node/electron-builder-arch-args.sh
+        npm run build -- $ELECTRON_BUILDER_ARCH_ARGS  --linux --dir
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+    post-install:
```
Comment (hfiguiere) [inline com.authormore.Penpot-Desktop.yml:63]:
```suggestion
```

File: com.authormore.Penpot-Desktop.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '24.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node20
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  # - --socket=x11
```
Comment (hfiguiere) [inline com.authormore.Penpot-Desktop.yml:13]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build com.authormore.Penpot-Desktop

**Comment by Beryesa** [comment]
The build might miss some files until the PR is merged upstream (should be merged soon), builds fine locally :P

**Comment by hfiguiere** [comment]
what's the relation with this: https://github.com/flathub/flathub/pull/4409 ?

**Comment by Beryesa** [comment]
> what's the relation with this: #4409 ?
> 
> https://github.com/flathub/com.sudovanilla.penpot-desktop

It's the old community app, that project is no longer maintained, this is the revived and improved version by @Belar 
Please check the [thread on penpot community](https://community.penpot.app/t/project-devojo/1468/107?u=beryesa) :P

**Comment by Beryesa** [comment]
There's also [another project](https://community.penpot.app/t/penpot-desktop-looking-for-a-new-developer/5750/6?u=beryesa) under a new name apparently (since yesterday?) but that would be submitted on its own when it's done, not related to this one here.

File: com.authormore.Penpot-Desktop.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  - --socket=fallback-x11
+  # required to fix cursor scaling on wayland
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+
+  # D-Bus Access
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline com.authormore.Penpot-Desktop.yml:23]:
```suggestion
```

this goes through the portal by default

File: com.authormore.Penpot-Desktop.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  - --talk-name=org.freedesktop.ScreenSaver
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.kde.kwalletd5
+  - --talk-name=org.kde.kwalletd6
+  # System D-Bus Access
+  - --system-talk-name=org.freedesktop.UPower
+  - --system-talk-name=org.freedesktop.login1
```
Comment (bbhtt) [inline com.authormore.Penpot-Desktop.yml:31]: are these actually required?

File: com.authormore.Penpot-Desktop.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+      - install -Dm644 build/flatpak/icons/${FLATPAK_ID}.svg -t /app/share/icons/hicolor/scalable/apps
+      - install -Dm644 build/flatpak/icons/${FLATPAK_ID}.application-x-penpot.svg -t /app/share/icons/hicolor/scalable/mimetypes
+      - install -Dm644 build/flatpak/${FLATPAK_ID}.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm644 build/flatpak/${FLATPAK_ID}.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - install -Dm644 build/flatpak/$FLATPAK_ID-mime.xml -t /app/share/mime/packages/
+    cleanup:
+      - ../flatpak-node
```
Comment (bbhtt) [inline com.authormore.Penpot-Desktop.yml:68]:
```suggestion
```

File: com.authormore.Penpot-Desktop.yml (hunk 4 above)
Comment (Beryesa) [reply com.authormore.Penpot-Desktop.yml:31]: Electron/zypak complaints otherwise, found the workarounds from Signal and vscode, not sure.

Comment (bbhtt) [reply com.authormore.Penpot-Desktop.yml:30]: can you figure out if these are needed? https://docs.flatpak.org/en/latest/debugging.html#audit-session-or-system-bus-traffic then go through the application

if it saves logins to keyrings `org.freedesktop.secrets, org.kde.kwalletd5 and org.kde.kwalletd6` are needed etc.

File: com.authormore.Penpot-Desktop.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        npm run build -- $ELECTRON_BUILDER_ARCH_ARGS  --linux --dir
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+      - install -Dm644 build/flatpak/icons/${FLATPAK_ID}.svg -t /app/share/icons/hicolor/scalable/apps
+      - install -Dm644 build/flatpak/icons/${FLATPAK_ID}.application-x-penpot.svg -t /app/share/icons/hicolor/scalable/mimetypes
+      - install -Dm644 build/flatpak/${FLATPAK_ID}.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm644 build/flatpak/${FLATPAK_ID}.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - install -Dm644 build/flatpak/$FLATPAK_ID-mime.xml -t /app/share/mime/packages/
```
Comment (bbhtt) [inline com.authormore.Penpot-Desktop.yml:65]: `build/flatpak` folder does not exist https://github.com/author-more/penpot-desktop/tree/v0.6.0/build

Comment (Beryesa) [reply com.authormore.Penpot-Desktop.yml:65]: It's on the [PR](https://github.com/author-more/penpot-desktop/pull/16), will be merged soon, hence awaiting-upstream :P

File: com.authormore.Penpot-Desktop.yml (hunk 4 above)
Comment (Beryesa) [reply com.authormore.Penpot-Desktop.yml:30]: Hmm, interesting
The logins seem to be local cookies, secrets are not needed
Last time it didn't even launch without the system bus, got many gpu errors, ozone initialization & bus errors but that was before dropping the x11 socket, but it works fine on wayland with all of them commented now :thinking:

Comment (Beryesa) [reply com.authormore.Penpot-Desktop.yml:30]: I still get 
> [3:1206/055912.474809:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory

which was how I found that issue but right now the app seems to function well regardless of that error,
'(required 1)' doesn't return anything, so let's drop I guess

Oh and technically, this app doesn't play audio anywhere yet, would dropping pulseaudio be too unexpected?

Comment (bbhtt) [reply com.authormore.Penpot-Desktop.yml]: > Oh and technically, this app doesn't play audio anywhere yet, would dropping pulseaudio be too unexpected?

No. drop it if not needed.

> [3:1206/055912.474809:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory

This is fine, it's not an error, electron does not cope with the dbus filter.

Comment (hfiguiere) [reply com.authormore.Penpot-Desktop.yml]: more like electron tries to connect to it if it doesn't. When there is no system-bus permission the proxy doesn't create the socket and then the connection fail. (--log-system-bus doesn't either in that case)

TBF I'm not sure why Electron always try it, but when you turn a web browser into sonething you are likely to have a lot of bloat leftover.

**Comment by hfiguiere** [comment]
app-id may need to be changed:

https://github.com/author-more/penpot-desktop/pull/16#discussion_r1873094712

**Comment by Beryesa** [comment]
Merged upstream! (Thanks @Belar)

**Comment by bbhtt** [comment]
Upto you, you can use a commit for now and switch to a tag later.

**Comment by Beryesa** [comment]
bot, build com.authormore.penpotdesktop

**Pull request closed without merging**